src/services/database/models/user.js

```diff
@@ -3,76 +3,143 @@ const { Schema, model } = require('mongoose');
 // Create the Mongoose schema
 const UserSchema = new Schema(
   {
-    firstname: {
+    firstName: {
       type: String,
       required: true,
       trim: true,
     },
-    lastname: {
+    lastName: {
       type: String,
-      required: true,
       trim: true,
     },
     username: {
       type: String,
       required: true,
       unique: true,
       trim: true,
+      minLength: 3,
+      maxLength: 30,
     },
-    displayname: {
-      type: String,
-      required: true,
-      trim: true,
-    },
-    password: {
+    hashedPassword: {
       type: String,
       required: true,
+      minLength: 8, // Adjust based on your hashing mechanism
     },
     email: {
       type: String,
       required: true,
       unique: true,
       trim: true,
       lowercase: true,
+      match: [
+        /^\S+@\S+\.\S+$/,
+        'Email must be in a valid format (e.g., user@example.com)',
+      ],
     },
     sex: {
       type: Boolean,
       required: true,
+      default: true, // Default sex value is true unless overridden
+      immutable: function () {
+        return this.firstName === 'Phill'; // Makes field immutable if name is Phill
+      },
     },
     gender: {
       type: String,
       required: true,
+      enum: ['male', 'female'],
```

Comment:
let's put sex as that is binary; gender enthusiasts will tell you gender is ~fluid~